src/store/index.js

```diff
@@ -846,15 +846,13 @@ function getStore(config, router) {
                 }
                 else if (typeof item.id !== 'undefined') {
                   let apiCollectionsLink = cx.getters.root?.getApiCollectionsLink();
-                  if (baseUrl) {
-                    url = Utils.toAbsolute(`items/${item.id}`, baseUrl);
-                  }
-                  else if (apiCollectionsLink) {
-                    url = Utils.toAbsolute(`${collectionId}/items/${item.id}`, apiCollectionsLink.href);
-                  }
-                  else if (cx.state.catalogUrl) {
+                  if (cx.state.catalogUrl) {
                     url = Utils.toAbsolute(`collections/${collectionId}/items/${item.id}`, cx.state.catalogUrl);
-                  }
```

Comment:
I don't agree with this regrouping. This part should only be a fallback, not the main path.

I did not check which of the other parts of the conditions it goes into, but if it's the "if baseUrl" then the issue is actually a server-side issue because there's a trailing slash missing in the self URL.

If it goes into "else if apiCollectionLink" then collectionId is not set and should probably be checked, so the condition should likely be `else if (apiCollectionsLink && collectionId)`.

Reply:
Indeed it is getting caught in the first condition, `if baseUrl`. The issue is we are using a server that strips trailing slashes implicitly in the response and relative links. Is it possible to ensure that `Utils.toAbsolute` can appropriately construct the URL regardless of if there is a trailing slash?

Reply:
No, that's not how resolving paths/URIs works. That's also not STAC-specific, that's an IT basic and will lead to issues in pretty much every STAC client that resolves the paths. We also have explained it in the STAC spec: https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#consistent-uris

You need to fix the server, there's no way around it. Alternative to adding trailing slashes is to add the last part of the path to the relative URLs.